### PR TITLE
fix: resolve failing issues with checkDSL validation on the new grammar

### PR DESCRIPTION
**Pull request by adriantam**
## Description
Correct DSL checker and update test snapshot to correct expected result.  In addition, add test to ensure dsl checker able to detect the right error line with empty line in DSL.

## References
Close https://github.com/openfga/syntax-transformer/issues/55

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: src/check-dsl.ts [hunk 1]
```diff
@@ -17,136 +20,106 @@ export const checkDSL = (codeInEditor: string) => {
...
-          return;
-        }
-        const lineIdx = typeDefinitionLine + idx + 1;
+      typeDef.relations.forEach((relationDef) => {
+        const { relation: relationName } = relationDef;
 
-        if (relations.includes(relation)) {
-          reporter.duplicateDefinition({ lineIndex: lineIdx, value: relation });
+        // Check if we have any duplicate relations
+        if (encounteredRelationsInType[relationName]) {
```
Comment (miparnisari) [inline src/check-dsl.ts:38]: how does this work? `encounteredRelationsInType` was defined in line 32 but you never added any values to it?

**Comment by miparnisari** [comment]
With this auth model, which has a bug:

```
type group
  relations
    define member as self
    define admin as self

    define can_add as can_manage_group  
    define can_edit as can_manage_group
    define can_delete as can_manage_group
    define can_read as can_manage
    
    define can_manage_group as admin
    define can_manage_users as admin
    define can_view_group as admin or member
    define can_view_users as admin or member
```

If I start typing another `define` at the end, I see this:

image

**Comment by adriantam** [comment]
> With this auth model, which has a bug:
> 
> ```
> type group
>   relations
>     define member as self
>     define admin as self
> 
>     define can_add as can_manage_group  
>     define can_edit as can_manage_group
>     define can_delete as can_manage_group
>     define can_read as can_manage
>     
>     define can_manage_group as admin
>     define can_manage_users as admin
>     define can_view_group as admin or member
>     define can_view_users as admin or member
> ```
> 
> If I start typing another `define` at the end, I see this:
> 
> image

Issue is fixed.  Thanks!

**Review by miparnisari — APPROVED** [review]
